### PR TITLE
Add enable_ip_forwarding option to azure_rm_networkinterface; Fixes #43276

### LINKED ISSUE #43276: ansible_rm_networkinterface should support IP forwarding
##### SUMMARY
`ansible_rm_networkinterface` should support IP forwarding. Especially important when deploying NVA VM's (i.e Palo Alto, F5 etc.)

##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
ansible_rm_networkinterface

##### ANSIBLE VERSION
```
2.6
```

Creating this issue in order to create branch for PR

### PR DESCRIPTION
**Pull request by spmp**
##### SUMMARY
Add `enable_ip_forwarding` to `azure_rm_networkinterface`

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
azure_rm_networkinterface

##### ANSIBLE VERSION
```
2.7
```

##### ADDITIONAL INFORMATION
Fixes #43276

### CODE REVIEW

**Comment by mattclay** [comment]
@spmp Could you update your PR to remove the style related changes?

Generally we prefer to avoid reformatting existing code to change things like indentation. It makes reviews and attribution of changes more difficult. It also tends to lead to back-and-forth edits for style related changes when they are not enforced by CI due to differences in contributor preferences.

**Comment by spmp** [comment]
@mattclay I made some changes to allow for alternative auth methods at the same time cleaned up inconsistent indenting. Can you live with this as it is a bit of work to revert changes then add back auth method etc. I have one more PR to get in then probably out of time for this 8(

**Comment by zikalino** [comment]
@spmp no worries, I am just doing some changes on another pr, will push today. i think this one needs rebase and resolving some conflicts.

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py [hunk 1]
```diff
@@ -185,6 +185,14 @@
             - When a default security group is created for a Linux host a rule will be added allowing inbound TCP
               connections to the default SSH port 22, and for a Windows host rules will be added allowing inbound
               access to RDP ports 3389 and 5986. Override the default ports by providing a list of open ports.
+    enable_ip_forwarding:
+        description:
+            - Whether to enable IP forwarding
+        aliases:
+            - ip_forwarding
```
Comment (yuwzho) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py:192]: Why adding an alias?

File: test/integration/targets/azure_rm_networkinterface/tasks/main.yml [hunk 2]
```diff
@@ -1,29 +1,34 @@
 - name: Prepare random number
   set_fact:
     rpfx: "{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}"
+    auth_source: auto
```
Comment (yuwzho) [inline test/integration/targets/azure_rm_networkinterface/tasks/main.yml]: `auth_source` should be inherit from `azure_rm_common` with `auto` value, no need write at this place

**Review by yuwzho — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py (hunk 1 above)
Comment (spmp) [reply lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py:192]: Why not? It seems like a reasonable alias. Can be removed if desired.

File: test/integration/targets/azure_rm_networkinterface/tasks/main.yml (hunk 2 above)
Comment (spmp) [reply test/integration/targets/azure_rm_networkinterface/tasks/main.yml]: I have these in my tests as I authenticate with `auth_source: cli` and change it back to `auto` before pushing. I could not find a common file for tests. I would be super happy to remove if there is a way to easily set this for testing. Using the other auth methods failed for me.

**Review by nitzmahone — CHANGES_REQUESTED** [review]
just need to fix the test indentation issues that are screwing up attribution, otherwise good to go

**Review by yuwzho — APPROVED** [review]

**Comment by yuwzho** [comment]
shipit

**Comment by spmp** [comment]
@nitzmahone Indentation changes reverted.

**Comment by yuwzho** [comment]
ready_for_review

**Review by nitzmahone — APPROVED** [review]

**Pull request merged**